packages/composites/market-status/src/transport/market-status.ts

```diff
@@ -0,0 +1,139 @@
+  async backgroundHandler(context: EndpointContext<BaseEndpointTypes>, entries: RequestParams[]) {
+    await Promise.all(entries.map(async (param) => this.handleRequest(context, param)))
+    await sleep(context.adapterSettings.BACKGROUND_EXECUTE_MS)
+  }
+
+  async handleRequest(context: EndpointContext<BaseEndpointTypes>, param: RequestParams) {
+    const requestedAt = Date.now()
+
+    let response: AdapterResponse<BaseEndpointTypes['Response']>
+    try {
+      const result = await this._handleRequest(context, param)
+      response = {
+        data: {
+          result: result.marketStatus,
+          source: result.source,
+        },
+        result: result.marketStatus,
+        statusCode: 200,
+        timestamps: {
+          providerDataRequestedUnixMs: requestedAt,
+          providerDataReceivedUnixMs: Date.now(),
+          providerIndicatedTimeUnixMs: result.providerIndicatedTimeUnixMs,
+        },
+      }
+    } catch (e) {
+      const errorMessage = e instanceof Error ? e.message : `Unknown error occurred: ${e}`
+      logger.error(e, errorMessage)
+      response = {
+        statusCode: 502,
+        errorMessage,
+        timestamps: {
+          providerDataRequestedUnixMs: requestedAt,
+          providerDataReceivedUnixMs: Date.now(),
+          providerIndicatedTimeUnixMs: undefined,
+        },
+      }
+    }
+    await this.responseCache.write(this.name, [{ params: param, response }])
+  }
+
+  async _handleRequest(
+    context: EndpointContext<BaseEndpointTypes>,
+    param: RequestParams,
+  ): Promise<MarketStatusResult> {
+    const market = param.market
+    if (!market) {
+      throw new Error(`Missing market in params: ${market}`)
+    }
+
+    const adapterNames = marketAdapters[market] ?? marketAdapters.__default
+
+    const primaryResponse = await this.sendAdapterRequest(context, adapterNames.primary, market)
+    if (primaryResponse.marketStatus !== MarketStatus.UNKNOWN) {
+      return primaryResponse
+    }
+
+    logger.warn(`Primary adapter ${adapterNames.primary} returned unknown market status`)
+
+    const secondaryResponse = await this.sendAdapterRequest(context, adapterNames.secondary, market)
+    if (secondaryResponse.marketStatus !== MarketStatus.UNKNOWN) {
+      return secondaryResponse
+    }
+
+    logger.error(
+      `Secondary adapter ${adapterNames.secondary} returned unknown market status, defaulting to CLOSED`,
+    )
+
+    return {
+      marketStatus: MarketStatus.CLOSED,
+      providerIndicatedTimeUnixMs: Date.now(),
+    }
+  }
+
+  async sendAdapterRequest(
+    context: EndpointContext<BaseEndpointTypes>,
+    adapterName: AdapterName,
+    market: string,
+  ): Promise<MarketStatusResult> {
+    const baseURL = context.adapterSettings[`${adapterName}_ADAPTER_URL`]
+    const data = {
+      data: {
+        endpoint: 'market-status',
+        market,
+      },
+    }
+    try {
+      const resp = await axios.post(baseURL, data, { timeout: 30_000 })
```

Comment:
We can initialize a requester instance and use it instead of axios
https://github.com/smartcontractkit/external-adapters-js/blob/main/packages/composites/gm-token/src/transport/price.ts#L57
It internally handles error retires, rate limits and request coalescing

Reply:
Update the PR!